octopoes/octopoes/models/ooi/config.py

```diff
@@ -9,7 +9,7 @@ class Config(OOI):
 
     ooi: Reference = ReferenceField(OOI)
     bit_id: str
-    config: dict[str, str]
+    config: dict
```

Comment:
It is possible to type this and the bit function arguments better if we want that. I don't know if we want config to be arbitrary JSON or we want to be more strict, but for arbitrary json the following type can be used:

`JSON: TypeAlias = dict[str, "JSON"] | list["JSON"] | str | int | float | bool | None`

See also https://github.com/python/typing/issues/182#issuecomment-1320974824 for more information

If we want config to always be a key-value pair where the value can be JSON we could use `dict[str, JSON]` as type. Although that might mean that we have to use casts or asserts in the bit code because mypy won't know about the json schema. If that gives too much problems another option might be `dict[str, Any]`.

@ammar92 any thoughts about what would be best?

Reply:
@dekkers I think this indeed is the best alternative, with the appropriate types and naming. I hope this will eventually end up as a builtin type in Python.

In many use cases they indeed use `dict[str, Any]`, but if we really must enforce JSON (without serialization) then the `JSON` type is the way to go. Although for me personally, the `dict[str, Any]` would also work, given that it has a good variable name, clear context or documented properly.

Reply:
Bit configs can be arbitrary json, where the schema is defined in the JsonSchema supplied in the related QuestionOOI.

Reply:
What is the outcome of this?

Reply:
@ammar92 @dekkers @underdarknl Everyone okay if I refactor to str, Any?

Reply:
Fine with me, we can always change this later if we want to.